Gems/Atom/Feature/Common/Code/Source/PostProcess/ColorGrading/HDRColorGradingSettings.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzCore/Serialization/SerializeContext.h>
+
+#include <PostProcess/ColorGrading/HDRColorGradingSettings.h>
+#include <PostProcess/PostProcessFeatureProcessor.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        HDRColorGradingSettings::HDRColorGradingSettings(PostProcessFeatureProcessor* featureProcessor)
+            : PostProcessBase(featureProcessor)
+        {
+        }
+
+        void HDRColorGradingSettings::OnConfigChanged()
+        {
+            m_parentSettings->OnConfigChanged();
+        }
+
+        void HDRColorGradingSettings::ApplySettingsTo(HDRColorGradingSettings* target, [[maybe_unused]] float alpha) const
+        {
+            AZ_Assert(target != nullptr, "HDRColorGradingSettings::ApplySettingsTo called with nullptr as argument.");
+
+            if (GetEnabled())
+            {
+                target->m_enabled = m_enabled;
+
+#define AZ_GFX_BOOL_PARAM(NAME, MEMBER_NAME, DefaultValue) ;
+#define AZ_GFX_FLOAT_PARAM(NAME, MEMBER_NAME, DefaultValue)                                                  \
+    {                                                                                                        \
+                target->Set##NAME(AZ::Lerp(target->MEMBER_NAME, MEMBER_NAME, alpha));                      \
+    }
+
+#define AZ_GFX_VEC3_PARAM(NAME, MEMBER_NAME, DefaultValue)                                                   \
+    {                                                                                                        \
+                target->MEMBER_NAME.Set(AZ::Lerp(target->MEMBER_NAME.GetX(), MEMBER_NAME.GetX(), alpha), \
+                    AZ::Lerp(target->MEMBER_NAME.GetX(), MEMBER_NAME.GetX(), alpha),                       \
+                    AZ::Lerp(target->MEMBER_NAME.GetX(), MEMBER_NAME.GetX(), alpha));                      \
```

Comment:
Is this correct? It's passing X for all three components. Should it be X, Y, Z? If it is supposed to be X, X, X then you only have to compute it once.

Reply:
Oops! Let me fix, it's meant to be the former.